tests/Models/BeatmapsetTest.php

```diff
@@ -255,243 +389,232 @@ public function testGlobalScopeSoftDelete()
     }
 
     public function testHybridNominateWithPlaymodePermissionMultipleModes(): void
     {
         $user = User::factory()->withGroup('bng', ['osu', 'taiko'])->create();
-        $beatmapset = $this->createHybridBeatmapset();
-
-        $notifications = Notification::count();
-        $userNotifications = UserNotification::count();
-
+        $beatmapset = $this->createHybridBeatmapset('taiko');
         $otherUser = User::factory()->create();
         $beatmapset->watches()->create(['user_id' => $otherUser->getKey()]);
 
-        $result = $beatmapset->nominate($user, ['osu', 'taiko']);
-
-        $this->assertTrue($result['result']);
-        $this->assertSame($notifications + 1, Notification::count());
-        $this->assertSame($userNotifications + 1, UserNotification::count());
-        $this->assertTrue($beatmapset->fresh()->isPending());
-
-        Bus::assertNotDispatched(CheckBeatmapsetCovers::class);
-    }
-
-    public function testHybridNominationBNGQualifyingBNGNominatedPartial(): void
-    {
-        $user = User::factory()->withGroup('bng_limited', ['osu', 'taiko'])->create();
-        $beatmapset = $this->createHybridBeatmapset();
-
-        $this->fillNominationsExceptLastForMode($beatmapset, 'bng', 'osu');
-        $this->fillNominationsExceptLastForMode($beatmapset, 'bng', 'taiko');
-
-        $result = $beatmapset->nominate($user, ['osu']);
+        $this->assertNotificationChanges();
+        $this->assertNominationChanges($beatmapset, ['osu', 'taiko']);
 
-        $this->assertTrue($result['result']);
-        $this->assertFalse($beatmapset->fresh()->isQualified());
+        $beatmapset->nominate($user, ['osu', 'taiko']);
 
+        $this->assertTrue($beatmapset->isPending());
         Bus::assertNotDispatched(CheckBeatmapsetCovers::class);
     }
 
-    public function testHybridNominationLimitedBNGQualifyingLimitedBNGNominated(): void
+    public function testQualifyingNominationBngLimited()
     {
-        $user = User::factory()->withGroup('bng_limited', ['osu', 'taiko'])->create();
         $beatmapset = $this->createHybridBeatmapset();
+        $beatmapset->nominate(User::factory()->withGroup('bng', ['osu', 'taiko'])->create(), ['osu', 'taiko']);
+        $nominator = User::factory()->withGroup('bng_limited', ['osu', 'taiko'])->create();
 
-        $this->fillNominationsExceptLastForMode($beatmapset, 'bng_limited', 'osu');
-        $this->fillNominationsExceptLastForMode($beatmapset, 'bng_limited', 'taiko');
+        $beatmapset->refresh();
 
-        $result = $beatmapset->fresh()->nominate($user, ['osu', 'taiko']);
+        $this->expectCountChange(fn () => $beatmapset->bssProcessQueues()->count(), 1);
 
-        $this->assertFalse($result['result']);
-        $this->assertSame($result['message'], osu_trans('beatmapsets.nominate.full_bn_required'));
-        $this->assertTrue($beatmapset->fresh()->isPending());
+        $beatmapset->nominate($nominator, ['taiko']);
```

Comment:
this test fails if it's `$beatmapset->fresh()->nominate(...)` instead. Replacing the refresh with `$beatmapset = $beatmapset->fresh()` doesn't help so the `bssProcessQueues()` maybe somehow do something to the nominate function? 🤷 

Also the nomination throws this error (not sure if it's tested somewhere in this file):

![](https://s.kohaku.love/2024/05/firefox_2024-05-30_19-11-06.png)

- map with taiko and fruits being eligible
- existing nominations (in order):
  1. fruits (bn)
  2. fruits (limited bn)
- trying to nominate (qualify) taiko (bn)

(I copied this test and noticed it doesn't fail unless I call fresh beforehand just like in this test)

Reply:
oh it was just the model used for assertion was different 🤔 still no idea why I'm getting that nomination error above though

Reply:
removing the `bssProcessQueues()` check doesn't have any affect 🤔